goal_src/levels/common/dark-eco-pool.gc

```diff
@@ -5,3 +5,1218 @@
+  (new 'static 'ripple-wave-set
+   :count 3
+   :converted #f
+   :normal-scale 3.0
+   :wave
+   (new 'static 'inline-array ripple-wave 4
+    (new 'static 'ripple-wave :scale 40.0 :xdiv 2 :speed 0.5)
+    (new 'static 'ripple-wave :scale 15.0 :xdiv -2 :zdiv 2 :speed 2.0)
+    (new 'static 'ripple-wave :scale 3.0 :xdiv 5 :zdiv 3 :speed 2.0)
+    (new 'static 'ripple-wave)
+    )
+   )
+  )
+
+;; definition for symbol ripple-for-maincave-dark-eco-pool, type ripple-wave-set
+(define
+  ripple-for-maincave-dark-eco-pool
+  (new 'static 'ripple-wave-set
+   :count 3
+   :converted #f
+   :normal-scale 8.57
+   :wave
+   (new 'static 'inline-array ripple-wave 4
+    (new 'static 'ripple-wave :scale 14.0 :xdiv 1 :speed 1.0)
+    (new 'static 'ripple-wave :scale 5.25 :xdiv -1 :zdiv 1 :speed 4.0)
+    (new 'static 'ripple-wave :scale 0.7 :xdiv 5 :zdiv 3 :speed 2.0)
+    (new 'static 'ripple-wave)
+    )
+   )
+  )
+
+;; definition for symbol ripple-for-finalboss-dark-eco-pool, type ripple-wave-set
+(define
+  ripple-for-finalboss-dark-eco-pool
+  (new 'static 'ripple-wave-set
+   :count 3
+   :converted #f
+   :normal-scale 4.0
+   :wave
+   (new 'static 'inline-array ripple-wave 4
+    (new 'static 'ripple-wave :scale 25.0 :xdiv 2 :speed -2.0)
+    (new 'static 'ripple-wave :scale 15.0 :xdiv -2 :zdiv 2 :speed 3.0)
+    (new 'static 'ripple-wave :scale 4.0 :xdiv 5 :zdiv 3 :speed 4.0)
+    (new 'static 'ripple-wave)
+    )
+   )
+  )
+
+;; definition for symbol ripple-for-dark-eco-pool, type ripple-wave-set
+(define
+  ripple-for-dark-eco-pool
+  (new 'static 'ripple-wave-set
+   :count 3
+   :converted #f
+   :normal-scale 3.0
+   :wave
+   (new 'static 'inline-array ripple-wave 4
+    (new 'static 'ripple-wave :scale 40.0 :xdiv 1 :speed 1.0)
+    (new 'static 'ripple-wave :scale 15.0 :xdiv -1 :zdiv 1 :speed 4.0)
+    (new 'static 'ripple-wave :scale 2.0 :xdiv 5 :zdiv 3 :speed 2.0)
+    (new 'static 'ripple-wave)
+    )
+   )
+  )
+
+;; definition for symbol ripple-for-sunken-dark-eco-helix-room, type ripple-wave-set
+(define
+  ripple-for-sunken-dark-eco-helix-room
+  (new 'static 'ripple-wave-set
+   :count 3
+   :converted #f
+   :normal-scale 8.0
+   :wave
+   (new 'static 'inline-array ripple-wave 4
+    (new 'static 'ripple-wave :scale 15.0 :xdiv 1 :speed 1.0)
+    (new 'static 'ripple-wave :scale 5.625 :xdiv -1 :zdiv 1 :speed 4.0)
+    (new 'static 'ripple-wave :scale 0.75 :xdiv 5 :zdiv 3 :speed 2.0)
+    (new 'static 'ripple-wave)
+    )
+   )
+  )
+
+;; definition for method 25 of type dark-eco-pool
+(defmethod TODO-RENAME-25 dark-eco-pool ((obj dark-eco-pool))
+  (let ((t9-0 (method-of-type water-anim TODO-RENAME-25)))
+   (t9-0 obj)
+   )
+  (set! (-> obj flags) (logand -8388609 (-> obj flags)))
+  (set! (-> obj flags) (logior #x1000000 (-> obj flags)))
+  (let
+   ((v0-1
+     ((the-as
+       (function res-lump symbol symbol float structure (pointer res-tag) pointer object)
+       (method-of-type res-lump get-property-struct)
```

Comment:
hm, this looks weird. Can you try again without a cast here? It should get turned into a `res-lump-struct` macro now.